receiver/elasticsearchreceiver/client.go

```diff
@@ -75,18 +77,30 @@ func newElasticsearchClient(settings component.TelemetrySettings, c Config, h co
 		authHeader = fmt.Sprintf("Basic %s", authb64)
 	}
 
-	return &defaultElasticsearchClient{
+	esClient := defaultElasticsearchClient{
 		client:     client,
 		authHeader: authHeader,
 		endpoint:   endpoint,
 		logger:     settings.Logger,
-	}, nil
+	}
+
+	// Try update es version
+	_, _ = esClient.ClusterMetadata(context.Background())
+	return &esClient, nil
 }
 
-// nodeStatsMetrics is a comma separated list of metrics that will be gathered from NodeStats.
-// The available metrics are documented here for Elasticsearch 7.9:
+// defaultNodeStatsMetrics: A comma separated list of metrics that will be gathered from NodeStats.
+// nodeStatsMetricsAfter7_9: Metrics only supported on and after 7.9
```

Comment:
style nit: typical in go constants are grouped together and comments are immediately above each item:

```go
const (
    // comment about foo
    foo = "foo"

    // comment about bar
    bar = "bar"
)
```

Reply:
good point, move all constants into a group.